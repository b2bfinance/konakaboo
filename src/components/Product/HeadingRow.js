import React from 'react';
import styled from 'styled-components';
import WhatsHotICon from '@material-ui/icons/Whatshot';
import Grid from '@material-ui/core/Grid';
import { Typography } from '@material-ui/core';
import Labels from './Labels';
import { queries } from '../../utils/media';

export const Wrapper = styled(Grid)`
  align-items: center;
  padding: 16px;
`;

export const HighlightedIcon = styled(WhatsHotICon)`
  margin-left: 0.25rem;
  color: ${props => props.theme.productHighlightColor};
`;

export const ProductLogoContainer = styled.div`
  background-color: white;
  margin-right: 16px;

  ${queries.desktop`
    border: 2px solid ${props => props.theme.productBorder};
    margin-bottom: -32px;
    padding: 8px;
    text-align: center;
  `};
`;

export const ProductLogo = styled.img`
  max-width: 100%;
  max-height: 125px;

  ${queries.desktop`
    max-height: 75px;
  `};
`;

export const ProductLabels = styled(Grid)`
  float: right;
`;

export default ({ logo, brand, highlighted, title, labels }) => (
  <Wrapper container>
    <Grid item xs={12} md={2}>
      <ProductLogoContainer>
        <ProductLogo src={logo} alt={brand} />
      </ProductLogoContainer>
    </Grid>
    <Grid item xs={12} md={7}>
      <Grid container>
        <Grid item xs={12}>
          <Grid container alignItems="center">
            <Grid item>
              <Typography variant="headline">{brand}</Typography>
            </Grid>
            <Grid item>{highlighted && <HighlightedIcon />}</Grid>
          </Grid>
        </Grid>
        <Grid item xs={12}>
          <Typography>{title}</Typography>
        </Grid>
      </Grid>
    </Grid>
    <Grid item xs={12} md={3}>
      {labels && labels.length > 0 && (
        <ProductLabels>
          <Labels labels={labels} />
        </ProductLabels>
      )}
    </Grid>
  </Wrapper>
);
