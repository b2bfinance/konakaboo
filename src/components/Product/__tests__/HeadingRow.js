import React from 'react';
import { render } from 'enzyme';
import HeadingRow, { ProductHeading } from '../HeadingRow';

describe('HeadingRow', () => {
  test('renders correctly without highlight', () => {
    expect(
      render(<HeadingRow labels={['foo', 'bar']} theme={stubData.theme} />)
    ).toMatchSnapshot();
  });

  test('renders correctly with highlight', () => {
    expect(
      render(
        <HeadingRow highlight labels={['foo', 'bar']} theme={stubData.theme} />
      )
    ).toMatchSnapshot();
  });
});

describe('ProductHeading', () => {
  test('renders correctly', () => {
    expect(render(<ProductHeading />)).toMatchSnapshot();
  });
});
