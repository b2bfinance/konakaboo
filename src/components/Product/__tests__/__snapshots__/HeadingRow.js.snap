// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`HeadingRow renders correctly with highlight 1`] = `
.c5 {
  border: 1px solid;
  border-radius: 1rem;
  font-size: 80%;
  padding: 0.2rem 0.75rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4:last-child div {
  margin-right: 0;
}

.c4 div {
  margin: .5rem .5rem 0 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.c1 {
  background-color: white;
  margin-right: 16px;
}

.c2 {
  max-width: 100%;
  max-height: 125px;
}

.c3 {
  float: right;
}

@media (min-width:36em) {
  .c4 {
    margin-left: auto;
  }

  .c4 div {
    margin: 0 .5rem 0 0;
  }
}

@media (min-width:62em) {
  .c1 {
    border: 2px solid;
    margin-bottom: -32px;
    padding: 8px;
    text-align: center;
  }
}

@media (min-width:62em) {
  .c2 {
    max-height: 75px;
  }
}

<div
  class="MuiGrid-container-1 c0"
>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-2-59"
  >
    <div
      class="c1"
    >
      <img
        class="c2"
      />
    </div>
  </div>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-7-64"
  >
    <div
      class="MuiGrid-container-1"
    >
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <div
          class="MuiGrid-container-1 MuiGrid-align-items-xs-center-9"
        >
          <div
            class="MuiGrid-item-2"
          >
            <h1
              class="MuiTypography-root-98 MuiTypography-headline-103"
            />
          </div>
          <div
            class="MuiGrid-item-2"
          />
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <p
          class="MuiTypography-root-98 MuiTypography-body1-107"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-3-60"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          foo
        </div>
        <div
          class="c5"
        >
          bar
        </div>
      </div>
    </div>
  </div>
</div>
`;

exports[`HeadingRow renders correctly without highlight 1`] = `
.c5 {
  border: 1px solid;
  border-radius: 1rem;
  font-size: 80%;
  padding: 0.2rem 0.75rem;
}

.c4 {
  display: -webkit-box;
  display: -webkit-flex;
  display: -ms-flexbox;
  display: flex;
  -webkit-flex-wrap: wrap;
  -ms-flex-wrap: wrap;
  flex-wrap: wrap;
}

.c4:last-child div {
  margin-right: 0;
}

.c4 div {
  margin: .5rem .5rem 0 0;
}

.c0 {
  -webkit-align-items: center;
  -webkit-box-align: center;
  -ms-flex-align: center;
  align-items: center;
  padding: 16px;
}

.c1 {
  background-color: white;
  margin-right: 16px;
}

.c2 {
  max-width: 100%;
  max-height: 125px;
}

.c3 {
  float: right;
}

@media (min-width:36em) {
  .c4 {
    margin-left: auto;
  }

  .c4 div {
    margin: 0 .5rem 0 0;
  }
}

@media (min-width:62em) {
  .c1 {
    border: 2px solid;
    margin-bottom: -32px;
    padding: 8px;
    text-align: center;
  }
}

@media (min-width:62em) {
  .c2 {
    max-height: 75px;
  }
}

<div
  class="MuiGrid-container-1 c0"
>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-2-59"
  >
    <div
      class="c1"
    >
      <img
        class="c2"
      />
    </div>
  </div>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-7-64"
  >
    <div
      class="MuiGrid-container-1"
    >
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <div
          class="MuiGrid-container-1 MuiGrid-align-items-xs-center-9"
        >
          <div
            class="MuiGrid-item-2"
          >
            <h1
              class="MuiTypography-root-98 MuiTypography-headline-103"
            />
          </div>
          <div
            class="MuiGrid-item-2"
          />
        </div>
      </div>
      <div
        class="MuiGrid-item-2 MuiGrid-grid-xs-12-41"
      >
        <p
          class="MuiTypography-root-98 MuiTypography-body1-107"
        />
      </div>
    </div>
  </div>
  <div
    class="MuiGrid-item-2 MuiGrid-grid-xs-12-41 MuiGrid-grid-md-3-60"
  >
    <div
      class="c3"
    >
      <div
        class="c4"
      >
        <div
          class="c5"
        >
          foo
        </div>
        <div
          class="c5"
        >
          bar
        </div>
      </div>
    </div>
  </div>
</div>
`;
